linera-views/src/reentrant_collection_view.rs

```diff
@@ -171,49 +199,46 @@ where
     pub async fn try_load_entry_mut(
         &mut self,
         short_key: Vec<u8>,
-    ) -> Result<RwLockWriteGuardArc<W>, ViewError> {
+    ) -> Result<WriteGuardedView<W>, ViewError> {
         *self.hash.get_mut() = None;
         let updates = self.updates.get_mut();
-        match updates.entry(short_key.clone()) {
-            btree_map::Entry::Occupied(entry) => {
-                let entry = entry.into_mut();
-                match entry {
-                    Update::Set(view) => Ok(view
-                        .clone()
-                        .try_write_arc()
-                        .ok_or_else(|| ViewError::TryLockError(short_key))?),
-                    Update::Removed => {
-                        let key = self
-                            .context
-                            .base_tag_index(KeyTag::Subview as u8, &short_key);
-                        let context = self.context.clone_with_base_key(key);
-                        // Obtain a view and set its pending state to the default (e.g. empty) state
-                        let mut view = W::load(context).await?;
-                        view.clear();
-                        let wrapped_view = Arc::new(RwLock::new(view));
-                        *entry = Update::Set(wrapped_view.clone());
-                        Ok(wrapped_view
-                            .try_write_arc()
-                            .ok_or_else(|| ViewError::TryLockError(short_key))?)
+        Ok(WriteGuardedView(
```

Comment:
Nit: use a let to decrease indentation

Reply:
We discussed this offline — this will be addressed in a later PR.